src/silx/gui/dialog/GroupDialog.py

```diff
@@ -164,6 +164,26 @@ def getSelectedDataUrl(self):
         """
         return self._selectedUrl
 
+    def setSelectedUrl(self, url: DataUrl):
+        """
+        Make the given url the selected URL. Raise an error if doesn't exists
+
+        :raises: ValueError if the url cannot be selected (file or data path not existing)
+        """
+        try:
+            self.addFile(path=url.file_path())
+        except OSError:
+            raise ValueError(f"File {url.file_path()} doesn't exists")
```

Comment:
If the file doesn't exists `addFile` will raise an `OSError`. Having the different use cases where a Url cannot be set make sense to me. But having an `OSError` would also make sense IMO

Reply:
No the error will be raised if we cannot add the file cannot be added (so if doesn't exists most likely)